src/editors/content/learning/contiguoustext/utils.tsx

```diff
@@ -312,17 +315,34 @@ export function getEntityAtCursor(editor: Editor): Maybe<Inline> {
   if (b === undefined) {
     return Maybe.nothing();
   }
-  const inner = b.nodes.get(s.anchor.path.get(1));
+  return Maybe.just(b);
+}
 
-  if (inner === undefined) {
-    return Maybe.nothing();
-  }
+export function getLeafAtCursor(editor: Editor): Maybe<Block | Text | Inline> {
```

Comment:
New function - selects the deepest leaf at the current cursor selection, which is used to target the foreign marks